CMakeLists_files.cmake

```diff
@@ -488,6 +492,8 @@ if(ENABLE_ECL_OUTPUT)
         opm/output/data/Cells.hpp
         opm/output/data/Solution.hpp
         opm/output/data/Wells.hpp
+        opm/output/eclipse/AggregateWellData.hpp
```

Comment:
We have no tradition for private (i.e. not installed/exported) headers, but I think we should start using it. The API is unwieldy and large - exposing many classes which are really only implementation details. The header files `AggregateWellData.hpp` and `CharArrayNullTerm.hpp` are in my opinion example of such headers. Opinions?


Mind: The opm-output module is *full* of headers which should have been private; I am by no mean singling out this PR as an offender in any way.

Reply:
Not installing headers is a logical thing to do for private headers, but is somewhat error prone unless installation is tested by Jenkins by default (I think it is not right now?) since changes can bring includes into a public header file easily, and requires some discipline. We should use this more, after ensuring that it gets tested properly.

However, it will not be of much use in ewoms or other heavily templated parts. We may want to separate these files as much for the developer/reader as for installation purposes. A simple way to signal this is to use a subdirectory `details` (for example) for such headers, that is at least used by some other projects/libraries.

Reply:
you think wrong. jenkins builds are purely installation based.